### DISCUSSION TITLE
Show documentation for macros?

### DISCUSSION THREAD

**Discussion by braxtons12** [clangd/clangd#779] [Q&A] [answered]
Is there a way to have documentation for a macro be shown in the hover/popup in the same way it would for types/typedefs/functions/etc?

if no, is this something that could be supported in the future?

I'm working on a C project that uses macros to approximate generics/templates and it would be really nice if the associated documentation could be seen in the hover and completion like it is for normal types/functions/etc.

**Comment by kadircet** [comment, marked as answer]
I've thought we had some discussions about this on https://github.com/clangd/clangd/issues/140 but maybe they were all offline.

The issue here is, comment extraction is completely handled by clang infrastructure and clangd merely formats the text and such. Unfortunately clang doesn't attach comments before a macro definition to that macro identifier, I am not familiar with those pieces of clang infra but I'd expect it to be doable.

**Reply by braxtons12** [reply]
Ah, okay.
Is that (1) something that someone on the clangd team could possibly submit to clang at some point, or (2) something that would be more reasonable to submit a (feature?) request against clang for?

if (2), would the right way to do that be to send an email to the cfe-dev list?

**Reply by Trass3r** [reply]
A bugzilla ticket can't hurt.

**Reply by kadircet** [reply]
the 2nd one definitely won't hurt in any case, and is sort of required no matter who ends up implementing it. because there's no clear owner for such parts of the codebase hence there might be some non-obvious reasons for not having the macro documentation support, and a discussion in cfe-dev would reveal that. so feel free to start a discussion in the mailing list.

**Reply by kadircet** [reply]
see https://github.com/llvm/llvm-project/issues/38206
